packages/aws-cdk-lib/aws-ecs/lib/base/service-managed-volume.ts

```diff
@@ -0,0 +1,190 @@
+   */
+  readonly volumeType?: ec2.EbsDeviceVolumeType;
+
+  /**
+   * The size of the volume in GiB.
+   *
+   * You must specify either `sizeInGiB` or `snapshotId`.
+   * You can optionally specify a volume size greater than or equal to the snapshot size.
+   *
+   * The following are the supported volume size values for each volume type.
+   *   - gp2 and gp3: 1-16,384
+   *   - io1 and io2: 4-16,384
+   *   - st1 and sc1: 125-16,384
+   *   - standard: 1-1,024
+   *
+   * @default - The snapshot size is used for the volume size if you specify `snapshotId`,
+   * otherwise this parameter is required.
+   */
+  readonly sizeInGiB?: number;
+
+  /**
+   * The snapshot that Amazon ECS uses to create the volume.
+   *
+   * You must specify either `sizeInGiB` or `snapshotId`.
+   *
+   * @default - No snapshot.
+   */
+  readonly snapShotId?: string;
+
+  /**
+   * The number of I/O operations per second (IOPS).
+   *
+   * For gp3, io1, and io2 volumes, this represents the number of IOPS that are provisioned
+   * for the volume. For gp2 volumes, this represents the baseline performance of the volume
+   * and the rate at which the volume accumulates I/O credits for bursting.
+   *
+   * The following are the supported values for each volume type.
+   *   - gp3: 3,000 - 16,000 IOPS
+   *   - io1: 100 - 64,000 IOPS
+   *   - io2: 100 - 256,000 IOPS
+   *
+   * This parameter is required for io1 and io2 volume types. The default for gp3 volumes is
+   * 3,000 IOPS. This parameter is not supported for st1, sc1, or standard volume types.
+   *
+   * @default - undefined
+   */
+  readonly iops?: number;
+
+  /**
+   * The throughput to provision for a volume, in MiB/s, with a maximum of 1,000 MiB/s.
+   *
+   * This parameter is only supported for the gp3 volume type.
+   *
+   * @default - No throughput.
+   */
+  readonly throughput?: number;
+
+  /**
+   * The Linux filesystem type for the volume.
+   *
+   * For volumes created from a snapshot, you must specify the same filesystem type that
+   * the volume was using when the snapshot was created.
+   * The available filesystem types are ext3, ext4, and xfs.
+   *
+   * @default - xfs.
+   */
+  readonly fileSystemType?: string;
+
+  /**
+  * Specifies the tags to apply to the volume and whether to propagate those tags to the volume.
+  *
+  * @default - No tags are specified.
+  */
+  readonly tagSpecifications?: EBSTagSpecification[];
+}
+
+/**
+ *  Tag Specification for EBS volume.
+ */
+export interface EBSTagSpecification {
+  /**
+  * The tags to apply to the volume.
+  *
+  * @default none
+  */
+  readonly tags?: {[key: string]: string};
+
+  /**
+  * Specifies whether to propagate the tags from the task definition or the service to the task.
+  * Valid values are: PropagatedTagSource.SERVICE, PropagatedTagSource.TASK_DEFINITION
+  *
+  * @default - undefined
+  */
+  readonly propagateTags?: PropagatedTagSource.SERVICE | PropagatedTagSource.TASK_DEFINITION;
```

Comment:
I thought the union type doesn't work for non-TypeScript code.

Reply:
https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md#unions

Reply:
Yeah, this should be an enum as well. @KollaAdithya

Reply:
changed `propagateTags` to an enum over here [[link]](https://github.com/aws/aws-cdk/pull/28691/files#diff-c9cc7970a22bbdf1add32b7adcb50af5f8bc6212db0340715d335b9446ef31e9R170-R183)